libs/ui-lib/lib/cim/components/Agent/AgentStatus.tsx

```diff
@@ -55,7 +55,7 @@ const AgentStatus: React.FC<AgentStatusProps> = ({
       AdditionalNTPSourcesDialogToggleComponent={AdditionalNTPSourcesDialogToggle}
       autoCSR
       additionalPopoverContent={
-        status.status.key === 'specSyncErr' ? (
+        status.status.key === 'spec-sync-err' ? (
```

Comment:
IMO the following change would make the code less error prone, and it should avoid future issues like the one you're solving now.

Changing the definition in  `helpers/agentStatus.tsx` file from

```
  specSyncErr: {
    key: 'spec-sync-err',
   ...
  },
```
to 
```
  specSyncErr: {
    key: 'specSyncErr',
   ...
  },
```

With this change, the object key and `key` value for this status would match, as it happens with all the other statuses.

Reply:
This is probably a good idea.

Reply:
Done.